.ci.yaml

```
@@ -3657,6 +3657,17 @@ targets:
       - bin/**
       - .ci.yaml
 
+  - name: Mac_ios native_platform_view_ui_tests_ios
+    bringup: true
+    recipe: devicelab/devicelab_drone
+    presubmit: false
+    timeout: 60
+    properties:
+      tags: >
+        ["devicelab", "ios", "mac"]
```

I led you astray, this isn't a devicelab test, which means it has a tethered iOS device, it should be "hostonly" since it only runs in the simulator.
```suggestion
    timeout: 60
    properties:
      tags: >
        ["devicelab", "hostonly"]
```

---

do you mean `["hostonly", "ios", "mac"]`?

---

No, I don't think so, I think that will make it require a machine with a tethered iOS device.
Like this:
https://github.com/flutter/flutter/blob/712860d33b00bf3acf02118512901a6261ad5219/.ci.yaml#L2394

---

@keyonghan The test is failed due to not having xcode isntalled:
https://ci.chromium.org/ui/p/flutter/builders/staging/Mac%20native_platform_view_ui_tests_ios/381/overview

```
xcrun: error: active developer path ("/opt/s/w/ir/cache/osx_sdk/XCode.app/Contents/Developer") does not exist
[native_platform_view_ui_tests_ios] [STDOUT] stdout:            Use `sudo xcode-select --switch path/to/Xcode.app` to specify the Xcode that you wish to use for command line developer tools, or use `xcode-select --install` to install the standalone command line developer tools.
[native_platform_view_ui_tests_ios] [STDOUT] stdout:            See `man xcode-select` for more details.
```
@jmagman pointed out that we may need "xcode" as dependency. Do you think we also need "gem" as dependencies? (similar to the next test `run_release_test_macos`): 

```
      dependencies: >-
        [
          {"dependency": "xcode", "version": "13a233"},
          {"dependency": "gems", "version": "v3.3.14"}
        ]
```

---

Could you try led runs locally with the xcode dependency? If it still fails, you can add the `gems` property as well to see if it helps.
```
led get-builder 'luci.flutter.staging:Mac native_platform_view_ui_tests_ios' | led edit -p 'xcode="13a233"' | led edit-recipe-bundle |  led launch
```

---

@keyonghan 
Got some result: 

Current state: https://chromium-swarm.appspot.com/task?id=5c076c0497ce4a10#
After adding xcode dep: https://chromium-swarm.appspot.com/task?id=5c080cac9d2b9d10
After adding both xcode and gem deps: https://chromium-swarm.appspot.com/task?id=5c0882ece20bbb10

Based on the time, looks like I will need to setup IAM in order to access the cloud log, but based on the timing, both changes failed likely due to bad setup.

---

Could you try updating the depot_tools?

Here is one run with both xcode and gems: https://ci.chromium.org/swarming/task/5c0901d5eda17d10?server=chromium-swarm.appspot.com
but it complains `This bot does not support codesigning`. Is the codesigning needed for this test? If yes, then it needs to run against a devicelab test bed.

---

This is the device lab testing so I guess it will require code signing for the app to run on devices, but if you just want to test the functionalities on simulator you can bypass code signing.

---

I think the `Mac` name before the test name indicates that it is to be tested on the simulator? @keyonghan is it correct?

---

Here the `devicelab` means tasks located under `dev/devicelab/bin/tasks`, which all share the same `devicelab_drone` recipes. It includes two types: the ones that need a devicelab testbed (which needs a physical phone connected), the other ones that run on a host only bot.

The `Mac` name will route all tests to the second type, whereas the `Mac_ios` will route tests to the first type.

The test added in this PR does run on a host only bot. You may want to take a look at the log to see if any findings: https://logs.chromium.org/logs/flutter/led/keyonghan_google.com/943f3399cb63bc7e1dc12ccdfcf310d85888f7de9f910789f46af2ed8a9f93cf/+/u/run_native_platform_view_ui_tests_ios/test_stdout

---

From the bottom of this link, it looks like the error is in Android: 
```
Task result:
{
  "success": false,
  "reason": "Task failed: DeviceException: The ANDROID_SDK_ROOT environment variable is missing. The variable must point to the Android SDK directory containing platform-tools."
}
```
It also mention about code signing tho, but not part of the failure reason.